hydrolib/core/io/dimr/models.py

```diff
@@ -218,52 +249,43 @@ class Parallel(BaseModel):
         start: Main component to be executed step wise (provides start time, end time and time step).
     """
 
+    _type: Literal["parallel"] = "parallel"
     startGroup: StartGroup
     start: ComponentOrCouplerRef
 
 
-class Control(BaseModel):
+class Start(ControlModel):
     """
-    Control flow specification for the DIMR-execution.
+    Specification of a serial control flow: one main component.
 
     Attributes:
-        parallel: Specification of a control flow that has to be executed in parallel.
-        start: Reference to the component instance to be started.
+        name: Name of the reference to a BMI-compliant model component instance
     """
 
-    parallel: Optional[List[Parallel]] = []
-    start: Optional[List[ComponentOrCouplerRef]] = []
-
-    @validator("parallel", "start", pre=True)
-    def validate_parallel(cls, v):
-        return to_list(v)
-
-    def is_intermediate_link(self) -> bool:
-        # TODO set to True once we replace Paths with FileModels
-        return False
+    _type: Literal["start"] = "start"
+    name: str
 
 
 class DIMR(FileModel):
     """DIMR model representation."""
 
     documentation: Documentation = Documentation()
-    control: Control = Control()
+    control: List[Union[Start, Parallel]] = Field(
+        [], discriminator="_type"
+    )  # used in Pydantic 1.9
     component: List[Union[RRComponent, FMComponent, Component]] = []
     coupler: Optional[List[Coupler]] = []
     waitFile: Optional[str]
     global_settings: Optional[GlobalSettings]
 
-    @validator("component", "coupler", pre=True)
+    @validator("component", "coupler", "control", pre=True)
     def validate_component(cls, v):
         return to_list(v)
 
     def dict(self, *args, **kwargs):
```

Comment:
nice!

Reply:
@evetion : what is the difference between `core.utils.to_list()` and `core.io.ini.util.make_list_validator()`? Both are used as validators. Should we keep both or not?